### PR TITLE
[UIKit] Added SnackbarUIView

### PR DESCRIPTION
**Pull request by LouisBorleeAdevinta**

### CODE REVIEW

File: Sources/Core/View/UIKit/SnackbarUIView.swift [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    private var imageContainer: UIView?
+    private let labelContainer = UIView()
+
+    private let horizontalStackView = UIStackView()
+    private let verticalStackView = UIStackView()
+
+    private let viewModel: SnackbarViewModel
+    private var cancellables = Set<AnyCancellable>()
+
+    @ScaledUIMetric private var scaleFactor: CGFloat = 1.0
```
Comment (robergro) [inline Sources/Core/View/UIKit/SnackbarUIView.swift:71]: I think it's better to use a @ScaledUIMetric for each properties and not a scaleFactor.
If you test for the cornerWidth, the scaled value is not equals to the cornerWidth * scaleFactor

Comment (LouisBorleeAdevinta) [reply Sources/Core/View/UIKit/SnackbarUIView.swift:71]: Done

File: Sources/Core/View/UIKit/SnackbarUIView.swift [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+        )
+        topAnchorConstraint.priority = .required - 1
+        topAnchorConstraint.isActive = true
+    }
+
+    /// Label trailing space management.
+    /// Spacing will be `0` if `button` exists and is in the `horizontalStackView`.
+    /// Spacing will be `(spacing.large - spacing.medium)` otherwise to be further away from the borders.
+    private func updateLabelTrailingConstraint() {
+        var trailingConstant = -(self.theme.layout.spacing.large - self.theme.layout.spacing.medium)
```
Comment (robergro) [inline Sources/Core/View/UIKit/SnackbarUIView.swift:191]: can be managed on the Viewmodel ..?

Comment (LouisBorleeAdevinta) [reply Sources/Core/View/UIKit/SnackbarUIView.swift:191]: Can't be done because UIKit and SwiftUI have different behaviours regarding the spacings

**Review by robergro — APPROVED** [review]

**Pull request merged**